plugin/src/main/java/org/opensearch/ml/action/stats/MLStatsNodesTransportAction.java

```diff
@@ -126,22 +146,72 @@ MLStatsNodeResponse createMLStatsNodeResponse(MLStatsNodesRequest mlStatsNodesRe
             }
         }
 
+        CountDownLatch latch = new CountDownLatch(mlStats.getAllModels().length);
         Map<String, MLModelStats> modelStats = new HashMap<>();
         // return model level stats
         if (mlStatsInput.includeModelStats()) {
             for (String modelId : mlStats.getAllModels()) {
-                if (mlStatsInput.retrieveStatsForModel(modelId)) {
-                    Map<ActionName, MLActionStats> actionStatsMap = new HashMap<>();
-                    for (Map.Entry<ActionName, MLActionStats> entry : mlStats.getModelStats(modelId).entrySet()) {
-                        if (mlStatsInput.retrieveStatsForAction(entry.getKey())) {
-                            actionStatsMap.put(entry.getKey(), entry.getValue());
+                // Add action listener to retrieve model details
+                validateAccess(modelId, ActionListener.wrap(hasPermissionToShowStat -> {
```

Comment:
so the change in here to show hidden model id if hasPermissionToShowStat, when I am the super user?

Reply:
Change here is: If it's superadmin user, only that time we will show the stats for hidden model. Other users will not be able to see the hidden model stats.